parallel/runner.go

```diff
@@ -237,17 +193,21 @@ func (r *runner) addThread() {
 	nextThreadId := atomic.AddUint32(&r.threadCount, 1) - 1
 	go func(threadId int) {
 		defer r.threadsWaitGroup.Done()
-
-		r.runningThreadsLock.Lock()
-		r.runningThreads++
-		r.runningThreadsLock.Unlock()
+		r.openThreadsLock.Lock()
+		r.openThreads++
```

Comment:
I suggest we use an atomic int here, and by doing so, remove `openThreadsLock ` altogether.

Reply:
I can use atomic int here, but I think that I still need the lock because of the if statement in line 229